streams/src/main/java/org/apache/kafka/streams/state/internals/NamedCache.java

```diff
@@ -200,6 +201,7 @@ synchronized void evict() {
         }
         remove(eldest);
         cache.remove(eldest.key);
+        dirtyKeys.remove(eldest.key);
```

Comment:
@guozhangwang 
on your first comment above about a`flush`. I don't see how that can happen.
In `evict` the item isn't removed from the cache until after `flush` has returned.

For your second comment - I think that could potentially happen. It should be fixed by always removing the key from ` dirtyKeys` in evict.

On the third comment. I don't think it matters about the ordering. The values for the keys that have been flushed will still be in the cache, so any store lookups for these values will result in cache hits.